elements/pfe-navigation/src/pfe-navigation.js

```diff
@@ -0,0 +1,431 @@
+    this.loginSlotMobile    = this.querySelector(`[slot="mobile-login"]`);
+    this.languageSlotMobile = this.querySelector(`[slot="mobile-language"]`);
+
+    ["search", "login", "language", "site-switcher"].forEach(slot => {
+      this.to_shadowdom(slot, `#pfe-navigation--${slot}`);
+    });
+
+    // Capture mobile slots from shadow
+    this._menuSlotMobile     = this.shadowRoot.querySelector(`slot[name="mobile-menu"]`);
+    this._searchSlotMobile   = this.shadowRoot.querySelector(`slot[name="mobile-search"]`);
+
+    if (this.children.length) {
+      this._initialized = this._init();
+    }
+
+    Promise.all([
+      customElements.whenDefined(PfeNavigationItem.tag),
+      customElements.whenDefined(PfeNavigationMain.tag)
+    ]).then(() => {
+      // Do a thing once items and main are loaded?
+      this._observer.observe(this, {
+        childList: true,
+        subtree: true,
+        characterData: true
+      });
+    });
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener(
+      `${PfeNavigationItem.tag}:toggled`,
+      this._toggledHandler
+    );
+
+    window.removeEventListener("scroll", this._stickyHandler);
+    document.removeEventListener("click", this._outsideListener);
+
+    this._observer.disconnect();
+  }
+
+  _observerHandler(mutationsList) {
+    // Reset the state to false, rerun the initializer
+    this._initialized = false;
+    this._initialized = this._init();
+  }
+
+  _toggledHandler(event) {
+    // If there is no active navigation item at the moment, open the clicked element
+    if (!this._activeNavigationItem && event.detail.navigationItem !== null) {
+      this._activeNavigationItem = event.detail.navigationItem;
+      // Add the overlay to the page
+      this.overlay.removeAttribute("hidden");
+      return;
+    }
+
+    // If the item clicked equals the currently active navigation item
+    if (this._activeNavigationItem === event.detail.navigationItem) {
+      // Close the navigation item and remove the overlay
+      this._activeNavigationItem = null;
+      this.overlay.setAttribute("hidden", true);
+      return;
+    }
+
+    // If no navigation item is provided
+    if (event.detail.navigationItem === null) {
+      // Close any open navigation items
+      this._activeNavigationItem.expanded = false;
+      // Set active to null and remove the overlay
+      this._activeNavigationItem = null;
+      this.overlay.setAttribute("hidden", true);
+      return;
+    }
+
+    // Otherwise, close the navigation item, open the next one
+    this._activeNavigationItem.expanded = false;
+    this._activeNavigationItem = event.detail.navigationItem;
+  }
+
+  _stickyHandler() {
+    if(window.pageYOffset >= this.top) {
+      this.classList.add("sticky");
+    } else {
+      this.classList.remove("sticky");
+    }
+  }
+
+  _outsideListener(event) {
+    if ((event.target !== this && event.target.closest("pfe-navigation") === null) || event.path[0] === this.overlay) {
```

Comment:
I was still coding on this - I did say it was WIP.  You pulled it down as I was coding this section.

Reply:
You pulled it down mid-coding so it wasn't ready yet.  It should be in a good place now.

Reply:
Yep, you noted it was ready for some preliminary branch testing; I was just noting UX issues where I thought it would be applicable in the MR (I tried not to add any comments specific to how code was written). I'll use a google doc next time